qa/wildfly/build.gradle

```diff
@@ -0,0 +1,190 @@
+        layout 'pattern', {
+            artifact 'wildfly/[revision]/[module]-[revision].[ext]'
+        }
+    }
+}
+
+configurations {
+    wildfly
+}
+
+dependencies {
+    providedCompile 'javax.enterprise:cdi-api:1.2'
+    providedCompile 'org.jboss.spec.javax.annotation:jboss-annotations-api_1.2_spec:1.0.0.Final'
+    providedCompile 'org.jboss.spec.javax.ws.rs:jboss-jaxrs-api_2.0_spec:1.0.0.Final'
+    compile ('org.jboss.resteasy:resteasy-jackson2-provider:3.0.19.Final') {
+        exclude module: 'jackson-annotations'
+        exclude module: 'jackson-core'
+        exclude module: 'jackson-databind'
+        exclude module: 'jackson-jaxrs-json-provider'
+    }
+    compile "com.fasterxml.jackson.core:jackson-annotations:${versions.jackson}"
+    compile "com.fasterxml.jackson.core:jackson-core:${versions.jackson}"
+    compile "com.fasterxml.jackson.core:jackson-databind:${versions.jackson}"
+    compile "com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:${versions.jackson}"
+    compile "com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:${versions.jackson}"
+    compile "com.fasterxml.jackson.module:jackson-module-jaxb-annotations:${versions.jackson}"
+    compile "org.apache.logging.log4j:log4j-api:${versions.log4j}"
+    compile "org.apache.logging.log4j:log4j-core:${versions.log4j}"
+    compile project(path: ':client:transport', configuration: 'runtime')
+    wildfly "org.jboss:wildfly:${wildflyVersion}@zip"
+    testCompile "org.elasticsearch.test:framework:${VersionProperties.elasticsearch}"
+}
+
+task unzipWildfly(type: Sync) {
+    into wildflyDir
+    from { zipTree(configurations.wildfly.singleFile) }
+}
+
+task deploy(type: Copy) {
+    dependsOn unzipWildfly, war
+    from war
+    into "${wildflyInstall}/standalone/deployments"
+}
+
+task writeElasticsearchProperties {
+    onlyIf { !Os.isFamily(Os.FAMILY_WINDOWS) }
+    dependsOn 'integTestCluster#wait', deploy
+    doLast {
+        final File elasticsearchProperties =
+                file("${wildflyInstall}/standalone/configuration/elasticsearch.properties")
+        elasticsearchProperties.write(["transport.uri=${-> integTest.getNodes().get(0).transportUri()}", "cluster.name=${-> integTest.getNodes().get(0).clusterName}"].join("\n"))
+    }
+}
+
+// the default configuration ships with IPv6 disabled but our cluster could be bound to IPv6 if the host supports it
+task enableIPv6 {
+    dependsOn unzipWildfly
+    doLast {
+        final File standaloneConf = file("${wildflyInstall}/bin/standalone.conf")
+        final List<String> lines =
+                Files.readAllLines(standaloneConf.toPath())
+                        .collect { line -> line.replace("-Djava.net.preferIPv4Stack=true", "-Djava.net.preferIPv4Stack=false") }
+        standaloneConf.write(lines.join("\n"))
+    }
+}
+
+task startWildfly {
+    dependsOn enableIPv6, writeElasticsearchProperties
+    doFirst {
+        final File script = new File(project.buildDir, "wildfly/wildfly.killer.sh")
```

Comment:
It doesn't *need* one because we [skip this test on Windows](https://github.com/elastic/elasticsearch/pull/24147/files#diff-b0d9c5d2992ba8499be25560312ae916R152). I do not see a good reason to jump through a lot of hoops to try to make it work on Windows. Please feel free to disagree and let me know if you think otherwise.